### DISCUSSION TITLE
Is Milvus an alternative for Redis?

### DISCUSSION THREAD

**Discussion by dinuthomas** [milvus-io/milvus#3923] [Q&A and General discussion] [answered]
Is Milvus an alternative for Redis? Please let me know if my understanding is correct.

Can I create the indexed vectors using Milvus and push it to redis to store the same (in the offline batch processing). The same vectors is to be retrieved during online serving through the Milvus library.

Or, Is Milvus a total replacement for Redis. If so, how to configure the Milvus data cache in a distributed cluster?

Basically we have a production live system which uses Redis for the key storage. The RAM size requirement is too huge. We are planning to move to Vector based approach and trying to use Milvus as a solution. I am looking for an architecture to easily pluggin Milvus into the current architecure to reduce the time to market. This is the background.

**Comment by shiyu22** [comment, marked as answer]
Hi,
Thanks for your sharing. 

Actually Milvus is a vector similarity search engine, unlike the traditional databases, Milvus currently only stores vectors and their corresponding ids, although we plan to store more information in future versions. So whether Milvus can totally replace Redis depends on your scenario, if the data only has vectors and ids then Milvus can. However, a lot of data contains more information than just ids, which requires access to a structured database such as Redis or MySQL.

I think Milvus can really help you if you're moving vector search, it's a great option! In distributed we use shared storage, you can refer to [Mishards](https://milvus.io/docs/v0.10.3/mishards.md), also [Milvus-Helm](https://github.com/milvus-io/milvus-helm).

**Reply by dinuthomas** [reply]
Thank you..

**Reply by CloudMarc** [reply]
Are there thoughts/plans to either a) add redis as an alternative to mysql for the milvus storage layer, and/or b) specify a "store" interface that can use mysql, redis, etc. as a plugin?

**Reply by yhmo** [reply]
> Are there thoughts/plans to either a) add redis as an alternative to mysql for the milvus storage layer, and/or b) specify a "store" interface that can use mysql, redis, etc. as a plugin?

We have no plan to do this. In fact, we will not provide major features on 1.0.x, but we will continue maintain it for critical bugs. As the announcement(https://github.com/milvus-io/milvus/discussions/4800) say, we are developing a new milvus, distributed, cloud-native, golang. All the new features will be considered/designed in the new milvus. The new milvus use ETCD as meta backend.

**Reply by yhmo** [reply]
Also, the new milvus provides stand-alone server.
